### PR TITLE
[BugFix] `SlotDescriptor` might be nullptr.

### PR DESCRIPTION
**Pull request by DorianZheng**
Signed-off-by: dorianzheng <xingzhengde72@gmail.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Review by Smith-Cruise — DISMISSED** [review]

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Smith-Cruise** [comment]
run starrocks_admit_test

File: be/src/formats/orc/orc_mapping.cpp [hunk 1]
```diff
@@ -108,6 +108,7 @@ Status OrcMappingFactory::_init_orc_mapping(std::unique_ptr<OrcMapping>& mapping
 
     for (size_t i = 0; i < slot_descs.size(); i++) {
         SlotDescriptor* slot_desc = slot_descs[i];
+        if (slot_desc == nullptr) continue;
```
Comment (imay) [inline be/src/formats/orc/orc_mapping.cpp:111]: Why can it be NULL? Should add some background and comments for this bug.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by DorianZheng** [comment]
run pr-title-check

**Comment by DorianZheng** [comment]
run pr-title-check

**Comment by DorianZheng** [comment]
@mergify backport 2.5

**Pull request merged**